editor/src/tool-bar/tool-bar.ts

```diff
@@ -237,15 +247,26 @@ export class ToolBar extends AbstractUIExtension implements IActionHandler, Edit
 
   onClickElementPickerToolButton = (button: HTMLElement, item: PaletteItem): void => {
     if (!this.editorContext.isReadonly) {
-      if (item.actions.length === 1 && item.id.startsWith('color-palette-item') && item.icon !== undefined) {
-        item.actions[0] = new ColorizeOperation([...this.selectionService.getSelectedElementIDs()], item.icon, item.label);
+      const actionKind = this.getActionKind(item.actions);
+      const selectedElementIds = [...this.selectionService.getSelectedElementIDs()];
+      if (actionKind === ColorizeOperation.KIND && item.icon !== undefined) {
```

Comment:
don't like this if...
I think we should here try to abstract this method, maybe by add the action as parameter?